src/virt.rs

```diff
@@ -751,7 +751,8 @@ impl VirtContext {
         Arch::write_csr(Csr::Medeleg, 0); // Do not delegate any exceptions
 
         self.csr.mie = Arch::read_csr(Csr::Mie);
-        self.csr.mip = Arch::read_csr(Csr::Mip);
+        self.csr.mip =
```

Comment:
Same remark here, it is not clear **why** it can be wrong. Someone new on the project can't deduce it from the comment (it is clearly a non-obvious part of the RISC-V spec).
Adding a bit more explanation about the why would help.